pkg/cleaner/cleaner.go

```diff
@@ -21,11 +21,19 @@ const (
 )
 
 type Ref struct {
-	mut *sync.Mutex
+	mut *sync.RWMutex
```

Comment:
@technosophos good question. I created the pass-through funcs so that, in theory, https://github.com/deis/builder/issues/192 could be done. It would involve a `map[string]*sync.RWMutex` (or similar) internally.